runtime/str.go

```diff
@@ -816,21 +816,26 @@ func strInterpolate(f *Frame, format string, values *Tuple) (*Object, *BaseExcep
 		if matches == nil {
 			return nil, f.RaiseType(ValueErrorType, "invalid format spec")
 		}
-		if matches[7] != "%" && valueIndex >= len(values.elems) {
+		flags, fieldWidth, fieldType := matches[1], matches[2], matches[7]
+		if fieldType != "%" && valueIndex >= len(values.elems) {
 			return nil, f.RaiseType(TypeErrorType, "not enough arguments for format string")
 		}
-		if matches[1] != "" {
-			return nil, f.RaiseType(NotImplementedErrorType, "conversion flags not yet supported")
-		}
-		if matches[2] != "" || matches[4] != "" {
-			return nil, f.RaiseType(NotImplementedErrorType, "field width not yet supported")
+		if (flags != "" && flags != "0") || fieldWidth == "*" || (fieldType != "d" && fieldType != "x" && fieldType != "X") {
```

Comment:
I would move the field width conversion out of the switch statement (since other types will use it eventually) and break up this large if condition a bit:

```
fieldWidth := -1
if matches[2] == "*" {
  return nil, f.RaiseType(...)
}
if matches[2] != "" {
  fieldWidth = .. Atoi conversion
}
```

Then in each field type block that doesn't support field width, raise when `fieldWidth != -1`. This will be more code but I think it makes it easier to reason about since each if statement has a very simple condition.